db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ArrayConstantNode.java

```diff
@@ -0,0 +1,407 @@
+/*
+ * Apache Derby is a subproject of the Apache DB project, and is licensed under
+ * the Apache License, Version 2.0 (the "License"); you may not use these files
+ * except in compliance with the License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Splice Machine, Inc. has modified this file.
+ *
+ * All Splice Machine modifications are Copyright 2012 - 2016 Splice Machine, Inc.,
+ * and are licensed to you under the License; you may not use this file except in
+ * compliance with the License.
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ */
+
+package com.splicemachine.db.impl.sql.compile;
+
+import com.splicemachine.db.iapi.reference.ClassName;
+import com.splicemachine.db.iapi.reference.SQLState;
+import com.splicemachine.db.iapi.services.classfile.VMOpcode;
+import com.splicemachine.db.iapi.services.sanity.SanityManager;
+import com.splicemachine.db.iapi.error.StandardException;
+import com.splicemachine.db.iapi.services.compiler.LocalField;
+import com.splicemachine.db.iapi.services.compiler.MethodBuilder;
+import com.splicemachine.db.iapi.sql.compile.Visitor;
+import com.splicemachine.db.iapi.types.DataTypeDescriptor;
+import com.splicemachine.db.iapi.types.TypeId;
+import com.splicemachine.db.iapi.util.JBitSet;
+import java.lang.reflect.Modifier;
+import java.sql.Types;
+import java.util.List;
+
+/**
+ * This node represents an Array Initialization Node.  It supports arrays in the format
+ * [1,1,1] and will set them in the corresponding row to a SQLArray
+ *
+ *
+ */
+
+public class ArrayConstantNode extends ValueNode {
```

Comment:
This class can be cleaned up a little bit, remove references to coalesce from comments if possible.